File: app/src/Domain/Type/FizzBuzzType2.php
<?php

namespace App\Domain\Type;

use App\Domain\Model\FizzBuzzValue;

class FizzBuzzType2 implements FizzBuzzType
{
    public function generate(int $number): FizzBuzzValue
    {
        return new FizzBuzzValue($number, (string) $number);
    }
}

File: app/fizz_buzz/lib/types/fizz_buzz_type2.ex
# FizzBuzzタイプ2の実装（5と7で異なるルール）
defmodule FizzBuzzType2 do
  use FizzBuzzTypeBase

  defimpl FizzBuzzType do
    def generate(_type, number) do
      FizzBuzzTypeBase.basic_fizzbuzz(number, 5, 7, "Fizz", "Buzz")
    end
  end
end